modules/agent/pkg/controllers/bundledeployment/controller.go

```diff
@@ -252,3 +277,113 @@ func readyError(status fleet.BundleDeploymentStatus) error {
 
 	return errors.New(msg)
 }
+
+func hasAutoReapply(bd *fleet.BundleDeployment, status *fleet.BundleDeploymentStatus) bool {
+	if bd.Spec.Options.Resync || bd.Spec.StagedOptions.Resync {
```

Comment:
The same condition has been checked before at https://github.com/rancher/fleet/pull/558/files#diff-8b83e3ec81ef037af3e68026d07535637ca1d84af565be9bcda91135c4b80714R177

Reply:
good spot. fixed it up.